### PR TITLE
Syntax highlight for public tests

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR is connected with #250 

### What's done:
- [x] Added RehypeHighlight plugin based on highlight.js
- [x] Added intellij-light.css
- [x] Added syntax highlight for public tests

This is how it looks like now:
screenshot

### CODE REVIEW

File: save-frontend/src/main/resources/highlight-css/intellij-light.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
+/*
+Intellij-light style (c) Pegasis <me@pegasis.site>
```
Comment (petertrr) [inline save-frontend/src/main/resources/highlight-css/intellij-light.css:2]: Why do we need code, when it's loaded as `<link>`?

File: save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt [hunk 2]
```diff
@@ -89,12 +89,17 @@ internal class ContestController(
...
             ?: return Mono.error(ResponseStatusException(HttpStatus.NOT_FOUND))
-        return TestFilesRequest(test.toDto(), testSuite.testRootPath).let {
+        return TestFilesRequest(test.toDto(), testSuite.testRootPath).let<TestFilesRequest, Mono<TestFilesContent>> {
             preprocessorWebClient.post()
                 .uri("/tests/get-content")
                 .bodyValue(it)
                 .retrieve()
                 .bodyToMono()
+        }.map { testFilesContent ->
+            testFilesContent.apply {
```
Comment (petertrr) [inline save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt:99]: I think it's better to make fields immutable and call `copy` here

Comment (sanyavertolet) [reply save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/ContestController.kt:99]: Yes, that is better

File: save-frontend/src/main/resources/highlight-css/intellij-light.css (hunk 1 above)
Comment (sanyavertolet) [reply save-frontend/src/main/resources/highlight-css/intellij-light.css:2]: Oh, will replace local css with css from cloudflare

**Review by petertrr — APPROVED** [review]

**Pull request merged**